<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MediaPlayerClass</class>
 <widget class="QWidget" name="MediaPlayerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>500</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>MediaPlayer</string>
  </property>
  <property name="windowIcon">
   <iconset resource="MediaPlayer.qrc">
    <normaloff>:/MediaPlayer/Resources/MediaPlayer.ico</normaloff>:/MediaPlayer/Resources/MediaPlayer.ico</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(0, 0, 0);</string>
  </property>
  <widget class="VideoWidget" name="videoWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>500</height>
    </rect>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(0, 0, 0);</string>
   </property>
  </widget>
  <widget class="QWidget" name="bottom_bar" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>440</y>
     <width>800</width>
     <height>60</height>
    </rect>
   </property>
   <property name="mouseTracking">
    <bool>false</bool>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgba(38, 76, 115, 150);</string>
   </property>
   <widget class="QWidget" name="control_bar" native="true">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>20</y>
      <width>200</width>
      <height>32</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgba(255, 255, 255, 0);</string>
    </property>
    <widget class="QPushButton" name="pushButton_stop">
     <property name="geometry">
      <rect>
       <x>8</x>
       <y>4</y>
       <width>24</width>
       <height>24</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="MediaPlayer.qrc">
       <normaloff>:/MediaPlayer/Resources/stop_24px.ico</normaloff>:/MediaPlayer/Resources/stop_24px.ico</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_previous">
     <property name="geometry">
      <rect>
       <x>46</x>
       <y>4</y>
       <width>24</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="MediaPlayer.qrc">
       <normaloff>:/MediaPlayer/Resources/previous_24px.ico</normaloff>:/MediaPlayer/Resources/previous_24px.ico</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_play">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>84</x>
       <y>0</y>
       <width>32</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="iconSize">
      <size>
       <width>32</width>
       <height>32</height>
      </size>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_next">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>4</y>
       <width>24</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="icon">
      <iconset resource="MediaPlayer.qrc">
       <normaloff>:/MediaPlayer/Resources/next_24px.ico</normaloff>:/MediaPlayer/Resources/next_24px.ico</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_sound">
     <property name="geometry">
      <rect>
       <x>168</x>
       <y>4</y>
       <width>24</width>
       <height>24</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">background-color: rgb(255, 255, 255);</string>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="Slider" name="horizontalSlider_media">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>800</width>
      <height>15</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="palette">
     <palette>
      <active>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="150">
          <red>38</red>
          <green>76</green>
          <blue>115</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="150">
          <red>38</red>
          <green>76</green>
          <blue>115</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="150">
          <red>38</red>
          <green>76</green>
          <blue>115</blue>
         </color>
        </brush>
       </colorrole>
      </active>
      <inactive>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="150">
          <red>38</red>
          <green>76</green>
          <blue>115</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="150">
          <red>38</red>
          <green>76</green>
          <blue>115</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="150">
          <red>38</red>
          <green>76</green>
          <blue>115</blue>
         </color>
        </brush>
       </colorrole>
      </inactive>
      <disabled>
       <colorrole role="Button">
        <brush brushstyle="SolidPattern">
         <color alpha="150">
          <red>38</red>
          <green>76</green>
          <blue>115</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Base">
        <brush brushstyle="SolidPattern">
         <color alpha="150">
          <red>38</red>
          <green>76</green>
          <blue>115</blue>
         </color>
        </brush>
       </colorrole>
       <colorrole role="Window">
        <brush brushstyle="SolidPattern">
         <color alpha="150">
          <red>38</red>
          <green>76</green>
          <blue>115</blue>
         </color>
        </brush>
       </colorrole>
      </disabled>
     </palette>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="focusPolicy">
     <enum>Qt::StrongFocus</enum>
    </property>
    <property name="contextMenuPolicy">
     <enum>Qt::DefaultContextMenu</enum>
    </property>
    <property name="maximum">
     <number>999</number>
    </property>
    <property name="pageStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>0</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="invertedAppearance">
     <bool>false</bool>
    </property>
    <property name="invertedControls">
     <bool>false</bool>
    </property>
    <property name="tickPosition">
     <enum>QSlider::NoTicks</enum>
    </property>
    <property name="tickInterval">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_playtime">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>25</y>
      <width>130</width>
      <height>20</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(255, 255, 255);
background-color: rgba(0, 0, 0, 0);
font: 11pt &quot;Arial&quot;;</string>
    </property>
    <property name="text">
     <string>00:00:00 / 00:00:00</string>
    </property>
   </widget>
  </widget>
  <widget class="QComboBox" name="comboBox_open">
   <property name="geometry">
    <rect>
     <x>350</x>
     <y>300</y>
     <width>100</width>
     <height>30</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">color: rgb(255, 255, 255);
background-color: rgba(38, 76, 115, 150);
font: 11pt &quot;Arial&quot;;</string>
   </property>
   <item>
    <property name="text">
     <string>打开文件</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>打开url</string>
    </property>
   </item>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>VideoWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>VideoWidget.h</header>
  </customwidget>
  <customwidget>
   <class>Slider</class>
   <extends>QSlider</extends>
   <header>Slider.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="MediaPlayer.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>pushButton_play</sender>
   <signal>clicked()</signal>
   <receiver>MediaPlayerClass</receiver>
   <slot>playOrPause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>388</x>
     <y>472</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>499</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_stop</sender>
   <signal>clicked()</signal>
   <receiver>MediaPlayerClass</receiver>
   <slot>stop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>324</x>
     <y>481</y>
    </hint>
    <hint type="destinationlabel">
     <x>242</x>
     <y>499</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboBox_open</sender>
   <signal>activated(QString)</signal>
   <receiver>MediaPlayerClass</receiver>
   <slot>openFileOrUrl()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>425</x>
     <y>316</y>
    </hint>
    <hint type="destinationlabel">
     <x>390</x>
     <y>499</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_media</sender>
   <signal>sliderPressed()</signal>
   <receiver>MediaPlayerClass</receiver>
   <slot>sliderPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>218</x>
     <y>452</y>
    </hint>
    <hint type="destinationlabel">
     <x>220</x>
     <y>499</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_media</sender>
   <signal>sliderReleased()</signal>
   <receiver>MediaPlayerClass</receiver>
   <slot>sliderReleased()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>445</y>
    </hint>
    <hint type="destinationlabel">
     <x>198</x>
     <y>499</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider_media</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>MediaPlayerClass</receiver>
   <slot>sliderMoved()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>454</y>
    </hint>
    <hint type="destinationlabel">
     <x>156</x>
     <y>499</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openFileOrUrl()</slot>
  <slot>playOrPause()</slot>
  <slot>sliderPressed()</slot>
  <slot>sliderReleased()</slot>
  <slot>stop()</slot>
  <slot>sliderMoved()</slot>
  <slot>volumeSliderPressed()</slot>
  <slot>volumeSliderReleased()</slot>
  <slot>volumeSliderMoved()</slot>
  <slot>volumeValueChanged()</slot>
  <slot>volumeActionTriggered()</slot>
 </slots>
</ui>
